### DISCUSSION TITLE
Adding data point type to ScriptableContext<'matrix'>

### DISCUSSION THREAD

**Discussion by kliyer-ai** [kurkle/chartjs-chart-matrix#76] [Q&A]
Hi there,

What's the best way to add the type of my datapoint to the context?

The type of my data points:
```ts
{
      x: '' + sameDay.toISOString(),
      y: '' + item.dow,
      d: `${weekDays[item.dow - 1]}, ${format(sameDay, 'h aaa')}`,
      v,
      raw: item,
 }
```

but I always get an error here:
```ts
backgroundColor(c: ScriptableContext<'matrix'>) {
                // @ts-expect-error ts problem
                const value = c.dataset.data[c.dataIndex]?.v ?? 0 // <- this errors!
                const alpha = value / mm
                return color('#2b988e').alpha(alpha).rgbString()
},
```

Any help would be appreciated. Thanks!!

**Comment by kurkle** [comment]
Should work the same as with other chart types: https://www.chartjs.org/docs/latest/api/classes/Chart.html

So something like `new Chart<'matrix', YourDatapointType>(...)`

I guess you could also augment the `MatrixDataPoint`type etc.
